### ISSUE TITLE
As a non-sighted user, I want to hear a description of visualizations embedded in essay content so that I can access these insights.

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/mep-django#404]
# dev notes
see [this article](https://www.sarasoueidan.com/blog/accessible-data-charts-for-khan-academy-2018-annual-report/) for an approach
- [x] alter the "embedded svg" block to use `<object>` tag, something like:
```html

<object
    role="img" 
    aria-label=“membership chart"
    type="image/svg+xml" data="chart.svg">

        <p>the logbooks show many more members than the cards. there are missing logbooks....etc etc</p>

</object>
```
- [x] add an optional "extended description" parameter that will render the inner text. clarify (via help text?) that this text is only read to non-sighted users and should describe the major insights or takeaways from the graphic. multiple paragraphs is ok

**Comment by thatbudakguy** [comment]
going to move this to next milestone, if no one objects.

**Comment by thatbudakguy** [comment]
I think I remember something about not being able to use `<object>` for this because wagtail doesn't allow it, but I'm a little confused about the markup that is in use now. For an example I created with @rlskoeser's bubble chart svg from PPA, I see:
```html
<figure role="img"
             aria-label="bubble graph of collections"
             aria-describedby="documents1collections-bubblesvg-desc">
    <img role="img"
              aria-hidden="true"
              src="/documents/1/collections-bubble.svg"
              alt="bubble graph of collections">
    <figcaption>
    <div class="rich-text"><p><i>Bubble chart of PPA collections showing relative sizes. Generated         with d3.js.</i></p></div>
     </figcaption>
     <div class="sr-only" id="documents1collections-bubblesvg-desc">
      <div class="rich-text"><p>This bubble chart shows that the Literary and Linguistic collections are the largest and very similar in size; Original Bibliography is the next largest but much smaller; the rest decrease in size, with Dictionaries as the smallest.</p>
      </div>
      </div>
</figure>
```
for this example in VoiceOver, the `<figcaption>` isn't read by or accessible to the screen reader, because the `<figure>` has an `aria-label` defined that overrides it and defines the accessible name for the entire `<figure>`. after a moment, however, the long description is read, because it's associated with the `<figure>`.

for the inner `<img>`, it has a `role` set to `img`, but is also `aria-hidden` despite having an `alt`. I don't really know what a defined role or alt would mean in the context of being `aria-hidden`, but at any rate they will never be used, as far as I know. again, this doesn't matter because the accessible name for the `<figure>` is defined via `aria-label` directly on it, so we wouldn't look to the `<img>`'s alt tag even if it were visible to an SR.

I tried a structure more like below, which I think felt a little more natural:
```html
<figure
    aria-describedby="documents2collections-bubblesvg-desc">
    <img src="/documents/2/collections-bubble.svg"
             alt="bubble graph of collections" role="img">
    <figcaption><div class="rich-text"><p><i>Bubble chart of PPA collections showing relative sizes. Generated with d3.js.</i></p></div></figcaption>
        <div class="sr-only" id="documents2collections-bubblesvg-desc"><div class="rich-text"><p>One problem with this chart is that it gives an inflated sense of the  PPA as a whole. Since numerous items are in multiple collections, they  are represented more than once. For comparison, look at a bubble chart  that includes the whole of the PPA among the collections. You can see  that the Literary and Linguistic collections each represent a sizable  proportion of the PPA, and that all the bubbles combined are much larger  than the entirety of the PPA.</p></div></div>
    
</figure>
```

for this, voiceover will initially read "Bubble chart of PPA collections showing relative sizes. Generated with d3.js. **Figure.**", which is similar to what a sighted user would see (the caption and its associated figure). after a moment, it will read the long description. crucially, if the user continues paging through the document via voiceover, they can then interact with the inner image (which will be announced as "bubble graph of collections, **image**" using the alt) and then finally with the description before leaving the figure.

I like this approach because it preserves the separate roles of caption, alt, and long description, and also allows the user to access the long description if they happened to not wait or miss it when they landed on the figure. Open to suggestions!

**Comment by rlskoeser** [comment]
@thatbudakguy thanks for the careful testing and markup suggestions, happy to revise. 

If I remember correctly, we had a concern was about the text inside the SVG being readable, which is why the img to be aria-hidden. Was this not a problem when you tested?

**Comment by thatbudakguy** [comment]
this wasn't a problem for me and I think is resolved by setting `role=img` on the svg, which we do (although not sure if there was any text in the svg I used anyway).
